components/script/dom/url.rs

```diff
@@ -105,6 +112,71 @@ impl URL {
     pub fn DomainToUnicode(_: GlobalRef, origin: USVString) -> USVString {
         USVString(domain_to_unicode(&origin.0))
     }
+
+    // https://w3c.github.io/FileAPI/#dfn-createObjectURL
+    pub fn CreateObjectURL(global: GlobalRef, blob: &Blob) -> DOMString {
+        if blob.IsClosed() {
+            // Generate a dummy id
+            let id = Uuid::new_v4().simple().to_string();
+            return DOMString::from(URL::unicode_serialization_blob_url(id));
+        }
+
+        let filemanager = global.resource_threads().sender();
+
+        // let origin = global.get_url().origin().unicode_serialization();
+        // XXX: Should we send this?
+
+        let slice = blob.get_slice_or_empty();
+        let bytes = slice.get_bytes();
+
+        let entry = BlobURLStoreEntry {
+            type_string: blob.Type().to_string(),
+            filename: None, // XXX: the filename is currently only in File object now
+            size: blob.Size(),
+            bytes: bytes.to_vec(),
```

Comment:
Yes, you are right. I haven't yet make `Blob` as powerful as it is expected to. But this field is necessary anyway.